File: app/src/main/java/com/waygo/data/provider/FlightStatusContract.java
package com.waygo.data.provider;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import com.waygo.data.base.contract.SerializedJsonContract;
import com.waygo.pojo.flightstatus.Flight;

import android.content.ContentValues;
import android.net.Uri;
import android.support.annotation.NonNull;

import java.lang.reflect.Type;

import rx.android.internal.Preconditions;

public class FlightStatusContract extends SerializedJsonContract<Flight> {
    private static final String TABLE_NAME = "flight_status";
    public static final Uri CONTENT_URI = Uri.parse("content://" + LufthansaContentProvider.PROVIDER_NAME + "/" + TABLE_NAME);
    private static final Type TYPE = new TypeToken<Flight>() {}.getType();

    @NonNull
    @Override
    public String getTableName() {
        return TABLE_NAME;
    }

    @NonNull
    @Override
    protected String getCreateIdColumn() {
        return ID + " VARCHAR(20)";
    }

    @NonNull
    @Override
    public ContentValues getContentValuesForItem(@NonNull Flight item) {
        Preconditions.checkNotNull(item, "Flight cannot be null.");

        ContentValues contentValues = new ContentValues();
        contentValues.put(ID, item.getId());
        contentValues.put(JSON, new Gson().toJson(item));
        return contentValues;
    }

    @NonNull
    @Override
    protected Type getType() {
        return TYPE;
    }
}

File: app/src/main/java/com/waygo/data/provider/FlightStatusContract.kt
package com.waygo.data.provider

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

import com.waygo.data.base.contract.SerializedJsonContract
import com.waygo.pojo.flightstatus.Flight

import android.content.ContentValues
import android.net.Uri

import java.lang.reflect.Type

import rx.android.internal.Preconditions

public class FlightStatusContract : SerializedJsonContract<Flight>() {

    override fun getTableName(): String =  TABLE_NAME

    override fun getCreateIdColumn(): String = SerializedJsonContract.ID + " VARCHAR(20)"

    override fun getType(): Type  = TYPE

    override fun getContentValuesForItem(item: Flight): ContentValues {
        Preconditions.checkNotNull(item, "Flight cannot be null.")

        val contentValues = ContentValues()
        contentValues.put(SerializedJsonContract.ID, item.getId())
        contentValues.put(SerializedJsonContract.JSON, Gson().toJson(item))
        return contentValues
    }



    companion object {
        private val TABLE_NAME = "flight_status"
        public val CONTENT_URI: Uri = Uri.parse("content://" + LufthansaContentProvider.PROVIDER_NAME + "/" + TABLE_NAME)
        private val TYPE = object : TypeToken<Flight>(){}.getType()
    }
}
